### PR TITLE
NN-3330 Turn on complexity functionality for select prisons

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: backend/config.js [hunk 1]
```diff
@@ -53,10 +53,10 @@ const apis = {
     timeoutSeconds: process.env.KEYWORKER_API_TIMEOUT_SECONDS || 30,
     ui_url: process.env.MANAGE_KEY_WORKERS_URL,
   },
-  complexityOfNeed: {
+  complexity: {
     url: process.env.COMPLEXITY_OF_NEED_URI || 'http://localhost:18091',
     timeoutSeconds: process.env.API_ENDPOINT_TIMEOUT_SECONDS || 10,
-    enabled: process.env.ENABLE_COMPLEXITY_API === 'true',
+    enabled_prisons: process.env.PRISONS_WITH_OFFENDERS_THAT_HAVE_COMPLEX_NEEDS,
```
Comment (sp-watson) [inline backend/config.js:59]: Hmm.. having this config in 2 separate services is error prone - never mind the fact that the complexity team probably have the same list somewhere else.

Also, we don't want to have to release the service just to add more prisons, but that is secondary.

File: backend/controllers/searchOffendersController.js [hunk 2]
```diff
@@ -52,7 +54,7 @@ module.exports = ({ allocationService, complexityOfNeedApi, keyworkerApi }) => {
         const otherKeyworkers = keyworkerResponse.filter((keyworker) => keyworker.staffId !== staffId)
 
         const isHighComplexity =
-          complexityOfNeed.enabled &&
+          isComplexityEnabledFor(activeCaseLoadId) &&
```
Comment (sp-watson) [inline backend/controllers/searchOffendersController.js:57]: as per psh, not sure the `isComplexityEnabledFor` call is needed?!

File: backend/controllers/viewResidentialLocation.js [hunk 3]
```diff
@@ -41,7 +43,7 @@ module.exports = ({ allocationService, elite2Api, keyworkerApi, complexityOfNeed
         const otherKeyworkers = keyworkerResponse.filter((keyworker) => keyworker.staffId !== offender.staffId)
         const formatNumberAllocated = (number) => (number ? `(${number})` : '')
         const isHighComplexity =
-          complexityOfNeed.enabled &&
+          isComplexityEnabledFor(activeCaseLoadId) &&
```
Comment (sp-watson) [inline backend/controllers/viewResidentialLocation.js:46]: as per psh, not sure the isComplexityEnabledFor call is needed?!

**Review by sp-watson — DISMISSED** [review]

File: backend/config.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/config.js:59]: It's technically in three if you include the key worker API. Other than storing the config in some shared persisted store I'm not sure what else to do, do you have any suggestions?

File: backend/controllers/viewResidentialLocation.js (hunk 3 above)
Comment (JamieDraperUK) [reply backend/controllers/viewResidentialLocation.js:46]: You're right it's not required, will fix now :+1:

File: backend/config.js (hunk 1 above)
Comment (sp-watson) [reply backend/config.js:59]: I will raise it with the tech team. I wasn't expecting any changes, just felt like i needed to comment on the problem?!

**Review by sp-watson — APPROVED** [review]

**Pull request merged**